oracle/price-feeder/oracle/jail.go

```diff
@@ -0,0 +1,31 @@
+package oracle
+
+const (
+	jailCacheInterval = int64(50)
+)
+
+type JailCache struct {
+	isJailed         bool
+	lastUpdatedBlock int64
+}
+
+func (jailCache *JailCache) Update(currentBlockHeight int64, isJailed bool) {
+	jailCache.lastUpdatedBlock = currentBlockHeight
+	jailCache.isJailed = isJailed
+}
+
+func (jailCache *JailCache) IsOutdated(currentBlockHeight int64) bool {
+	if currentBlockHeight < jailCacheInterval {
+		return false
+	}
+
+	// This is an edge case, which should never happen.
+	// The current blockchain height is lower
+	// than the last updated block, to fix we should
+	// just update the cached params again.
+	if currentBlockHeight < jailCache.lastUpdatedBlock {
+		return true
+	}
```

Comment:
Hmm, if it should never happen then should this be a panic or at least log an error?

Reply:
Yeah, we don't really need this and the above check. It should only become stale if currentBlockHeight > lastUpdate + interval